gitlab/v4/objects/projects.py

```diff
@@ -175,6 +178,7 @@ class Project(RefreshMixin, SaveMixin, ObjectDeleteMixin, RepositoryMixin, RESTO
     pagesdomains: ProjectPagesDomainManager
     pipelines: ProjectPipelineManager
     pipelineschedules: ProjectPipelineScheduleManager
+    protectedenvironments: ProjectProtectedEnvironmentManager
```

Comment:
Sure. 

At some point you might want to proxy unmatching attributes to keep a naming coherence ?

```python
protected_environments: ProjectProtectedEnvironmentManager
protectedenvironments: self.protected_environments  # or anything that actually works
```

to avoid surprise as in 

```python
print(project.protected_environments, project.protectedvariables)
```

we might still ensure identity 

```
assert project.protected_environments is project.protectedenvironments
```.

Anyway I'm taking the freedom to resolve this conversation :)

Reply:
@calve missed this one - yes that definitely makes sense for some of our other managers too, we did something similar for the breaking changes in v3.0.0 but should probably check all of our endpoints again. Thanks @JohnVillalovos for pointing this out :)